src/game/default/g_ballistics.c

```diff
@@ -354,6 +418,40 @@ static void G_GrenadeProjectile_Touch(g_entity_t *self, g_entity_t *other,
 /*
  * @brief
  */
+void G_HandGrenadeProjectile_Touch(g_entity_t *self, g_entity_t *other,
```

Comment:
Again, this could probably be the same function for regular grenades, with any special behavior conditionalized on self->class_name.